core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1415,10 +1415,16 @@ package object config {
 
   private[spark] val SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED =
     ConfigBuilder("spark.shuffle.readHostLocalDisk")
-      .doc(s"If enabled (and `${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key}` is disabled and external " +
-        s"shuffle `${SHUFFLE_SERVICE_ENABLED.key}` is enabled), shuffle " +
-        "blocks requested from those block managers which are running on the same host are read " +
-        "from the disk directly instead of being fetched as remote blocks over the network.")
+      .doc("When enabled, shuffle blocks requested from those block managers which are running " +
+        "on the same host are read from the disk directly instead of being fetched as remote " +
+        "blocks over the network. Note that for k8s workloads, this only works when nodes are " +
+        "using non-isolated container storage." +
+        s"To enable the feature, one should disable ${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key} first." +
+        " And make sure that one of the following requirements are satisfied:\n" +
+        s"1. external shuffle service is enabled (${SHUFFLE_SERVICE_ENABLED.key});" +
+        s"2. dynamic allocation is disabled (${DYN_ALLOCATION_ENABLED.key});" +
+        s"3. dynamic allocation is enabled with shuffle tracking " +
+        s"(${DYN_ALLOCATION_SHUFFLE_TRACKING_ENABLED});")
```

Comment:
Please take a look at the updated document regarding the dynamic allocation. cc: @holdenk @attilapiros

Reply:
To enable dynamic allocation, it requires spark.shuffle.service.enabled or spark.dynamicAllocation.shuffleTracking.enabled. So I think users must be in one among the 3 conditions.

Looks like these requirements are redundant?

Reply:
I agree. I think we should take these out. In the future we may have different requirements and forget to go change here as well.

Reply:
I agree with @viirya and @holdenk.
 
@Ngone51 what would be the case when this feature is switched on (and SHUFFLE_USE_OLD_FETCH_PROTOCOL is disabled) but it cannot be used? If there is maybe with that the description can be simplified.

Reply:
Oh yeah... we'd cover all the cases after this PR. I'll remove the requirements.